extras/ci/analyze-ci-parallelization.py

```
@@ -0,0 +1,344 @@
+    parallelization_efficiency = None
+    if total_duration > 0:
+        parallelization_efficiency = total_cpu_time / total_duration
+
+    # Check overall efficiency
+    if parallelization_efficiency is None:
+        print("\nPARALLELIZATION EFFICIENCY UNAVAILABLE")
+        print("   Workflow duration is zero; cannot compute efficiency.")
+    elif parallelization_efficiency < 5:
+        print("\nLOW PARALLELIZATION EFFICIENCY")
+        print(f"   Current: {parallelization_efficiency:.1f}x")
+        print(f"   Many jobs are running sequentially. Consider:")
```

Ruff (0.15.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)